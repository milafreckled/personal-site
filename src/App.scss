@mixin printText($num-letters: 3) {
  //$angle-per-char: $angle-span / $num-letters;
  @for $i from 1 through $num-letters {
    &:nth-of-type(n) {
      animation: 3s appear;
      animation-delay: $i+'s';
    }
  }
}
.scroll-lock{
  overflow: hidden;
  margin-right: -17px;
}
body, * {
    overflow-x: hidden;
    scroll-behavior: smooth;
    margin: 0;
  }
  h6, h2, h4{
    margin-block-start: 0;
    margin-block-end: 0;
  }
  a{
    text-decoration: none;
    color: "#fff";
    background-color: transparent;
  }
  a:visited, a:hover, a:active{
    color: "#fff";
    background-color: transparent;
  }
  .glow-icon{
    border-radius: 50%;
  }
  .glow-icon::before{
content: '';
position: absolute;
top: 0;
left: 0;
width: 60%; 
height: 60%;
border-radius: 50%;
background: white;
transition: .5s;
transform: scale(.9);
z-index: -1;
  }
  .glow-icon:hover::before{
    transform: scale(1.1);
    box-shadow: 0 0 15px white;
  }
  .glow-icon:hover{
    box-shadow: 0 0 5px white;
    text-shadow: 0 0 5px white;
  }
  .MuiPaper-root, .MuiPaper-elevation, .MuiPaper-rounded{
    position: "relative";
    text-align: 'center';
    display: 'flex';
    align-items: 'center';
    justify-content: 'center';
    flex-direction: 'column';
    color: black;
    width: "30vw";
    height: "20vw";
    background-position: "cover";
    background-size: "30rem";
   &:hover{
     background-image: 'Å†one';
   }
    >.project-title{
      position: relative;
      background-color: 'black';
      text-align: center;
      align-items: center;
      width: 100%;
      height: 100%;
      display: block;
      z-index: '4';
      text-transform: uppercase;
    >*{
      padding: '30%';
    }
    }
  }   

  .project-stack{
    min-height: "20vw";
  }

  ul{
    list-style: none;
    position: relative;
    width: 100%;
  }
  span{
    position: absolute;
    width: 3em;
    height: 3em;
    left: -1.5em;
    top: 0;
    transition: all .4s;
    @media only screen and (max-width: 768px){
      left: -2em;
    }
  }
  li{
    position: relative;
    display: block;
    overflow: visible;
    width: 100%;
  }
  li:hover {
    transform: translateZ(1.5rem);
 }
li:hover #emoji-rocket{
     animation: ease-in-out fly 1.5s forwards;
  }
li:hover #emoji-teamplayer{
    animation: ease-in-out teamplayer 1.5s forwards;
 }
 li:hover #emoji-curious{
  animation: ease-in-out curious 1.5s forwards;
}
li span{
  cursor: pointer;
}
  @keyframes fly{
    50%{
     transform: translate(50%, -50%);
    }
  }

 @keyframes curious{
   50%{
    transform: translateX(-2em);
   }
   100%{
    transform: scaleX(1.1);
   }
  }

 @keyframes teamplayer{
   50%{
    transform: scale(1.2);
   }
  }
  .justHi{
    @include printText;
    color: #4B3869;
  }

  @keyframes appear{
    from{
      opacity: 0;
    }
    to{
      opacity: 1;
    }
  }

  .MuiSvgIcon-root {
    fill: #fff;
  }
 
  h4.rotated-text{
   transform-origin: bottom center;
    font: 26px, 'Inconsolata';
  }


  @keyframes slideInFromTop {
    0% {
      opacity: 0.6;
    }
    100% {
      opacity: 1;
    }
  }
  @keyframes fadeUp{
  0%{
    opacity: 0;
    transform: translateY(50%);
  }
  100%{
    opacity: 1;
    transform: translateY(0);
  }
  }
  @keyframes appear{
    from{
      opacity: 0; 
    }
    to{
      opacity: 100%;
    }
}
  
@keyframes blink{
  from{
      box-shadow: 0px 0px 10px #4B3869;  
  }
  to{
    box-shadow: 0px 0px 20px rgba(134, 52, 172, 0.6);
  }
}
