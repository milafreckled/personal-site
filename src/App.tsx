import React from 'react';
import Navigation from './Components/Navigation';
import Portrait  from './Components/Portrait'
import "./App.scss"
import myTheme from './Theme/MyTheme'
import { ThemeProvider } from '@mui/material/styles';
import  Projects  from "./Components/Projects"
import Timeline from "./Components/Timeline"
import Footer from "./Components/Footer"

function App() {
  return (
 <ThemeProvider theme={myTheme}>
     <Navigation></Navigation>
    <Portrait />
   <Projects></Projects>
   <Timeline></Timeline>
   <Footer></Footer>
    </ThemeProvider>
  );
}

export default App;
