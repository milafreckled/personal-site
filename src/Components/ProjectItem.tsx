import * as React from 'react';
import Paper from '@mui/material/Paper';
import "../App.scss"
import { makeStyles } from '@mui/styles'
import myTheme from '../Theme/MyTheme'

const useStyles=makeStyles({
  projectItem: {
    margin: '.5vw 2.5vw',
    padding: '.2em',
    position: 'relative',
    borderRadius: '.1em',
    color: myTheme.palette.primary.main,
    backgroundColor: '#fff', 
    letterSpacing: '.3rem',
    transition: 'all .4s ease-in-out',
    textTransform: 'uppercase',
    textAlign: 'center',
    perspective: '500px',
    transformStyle: 'preserve-3d',
    display: 'block',
    minWidth: '45vw',
    '& :hover':{
      boxShadow: `2px 2px 10px ${myTheme.palette.text.disabled}`,
      transform: 'translateZ(2rem)',
      letterSpacing: '2.5',
    },
    [myTheme.breakpoints.down('sm')]:{
      margin: '1vw 2.5vw',
    }
  },

})
interface ProjectItemProps {
    text: string, 
    url: string, 
 }

export default function ProjectItem(props: ProjectItemProps){
    const classes = useStyles()
    return(
      <a className={classes.projectItem} href={props.url}>{props.text}</a>

    )
}