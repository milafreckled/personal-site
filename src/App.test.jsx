import * as React from 'react';
import { getByTestId, render, screen } from '@testing-library/react';
import App from './App';
import renderer, { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import ProjectItem from './Components/ProjectItem';
import weatherAPI from './images/weather-api.jpeg'
import shallow from 'react-test-renderer/shallow'; 

test('renders project item component', () => {

  const component = renderer.create(
    <ProjectItem url="https://thawing-beach-25991.herokuapp.com/" bgImage={`url(${weatherAPI})`} text="Weather API"></ProjectItem>
  ); 
  const result = component.getRenderOutput();
  let tree = component.toJSON();
  let Href = tree.props.url;
  console.log(component.getInstance());
  expect(tree).toMatchSnapshot();
  const wrapper = shallow(component);
  expect(result.type).toBe(`div`)
});

