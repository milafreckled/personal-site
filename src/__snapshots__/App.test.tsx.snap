// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="makeStyles-projectItem-1"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "height": "20vw",
      "position": "relative",
      "width": "30vw",
    }
  }
>
  <div
    style={
      Object {
        "background": "url(weather-api.jpeg)",
        "backgroundSize": "contain",
        "filter": "none",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <a
    className="makeStyles-projectLink-3"
    href="https://thawing-beach-25991.herokuapp.com/"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Weather API
  </a>
</div>
`;

exports[`renders project item component 1`] = `
<div
  className="makeStyles-projectItem-1"
  data-testid="parent"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "height": "20vw",
      "position": "relative",
      "width": "30vw",
    }
  }
>
  <div
    style={
      Object {
        "background": "url(weather-api.jpeg)",
        "backgroundSize": "contain",
        "filter": "none",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <a
    className="makeStyles-projectLink-3"
    href="https://thawing-beach-25991.herokuapp.com/"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Weather API
  </a>
</div>
`;
